core/qbft/qbft.go

```diff
@@ -202,6 +203,29 @@ func Run[I any, V comparable](ctx context.Context, d Definition[I, V], t Transpo
 		dedupIn = dedup
 	}
 
+	isDuplicatedRule := func(rule uponRule) bool {
+		if rule == uponJustifiedDecided {
+			return false
+		}
+		if dedupRules[rule] {
+			return true
+		}
+
+		// first time for this rule
```

Comment:
nit (also please capitalise comment sentences): `// First time for this rule in current round`

Reply:
Done